provisioning/provisioning-service-client/src/main/java/com/microsoft/azure/sdk/iot/provisioning/service/implementation/ProvisioningServiceClientImpl.java

```diff
@@ -0,0 +1,1823 @@
+import com.microsoft.rest.ServiceClient;
+import com.microsoft.rest.RestClient;
+import okhttp3.OkHttpClient;
+import retrofit2.Retrofit;
+import com.google.common.reflect.TypeToken;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.AttestationMechanism;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.BulkEnrollmentOperation;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.BulkEnrollmentOperationResult;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.DeviceRegistrationState;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.EnrollmentGroup;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.IndividualEnrollment;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.ProvisioningServiceErrorDetailsException;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.QueryEnrollmentGroupsHeaders;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.QueryIndividualEnrollmentsHeaders;
+import com.microsoft.azure.sdk.iot.provisioning.service.models.QuerySpecification;
+import com.microsoft.rest.ServiceCallback;
+import com.microsoft.rest.ServiceFuture;
+import com.microsoft.rest.ServiceResponse;
+import com.microsoft.rest.ServiceResponseWithHeaders;
+import com.microsoft.rest.Validator;
+import java.io.IOException;
+import java.util.List;
+import okhttp3.ResponseBody;
+import retrofit2.http.Body;
+import retrofit2.http.GET;
+import retrofit2.http.Header;
+import retrofit2.http.Headers;
+import retrofit2.http.HTTP;
+import retrofit2.http.Path;
+import retrofit2.http.POST;
+import retrofit2.http.PUT;
+import retrofit2.http.Query;
+import retrofit2.Response;
+import rx.functions.Func1;
+import rx.Observable;
+
+/**
+ * Initializes a new instance of the ProvisioningServiceClient class.
+ */
+public class ProvisioningServiceClientImpl extends ServiceClient implements ProvisioningServiceClient {
+    /**
+     * The Retrofit service to perform REST calls.
+     */
+    private ProvisioningServiceClientService service;
+
+    /** The API version to use for the request. Supported versions include: 2018-09-01-preview. */
+    private String apiVersion;
+
+    /**
+     * Gets The API version to use for the request. Supported versions include: 2018-09-01-preview.
+     *
+     * @return the apiVersion value.
+     */
+    public String apiVersion() {
+        return this.apiVersion;
+    }
+
+    /**
+     * Sets The API version to use for the request. Supported versions include: 2018-09-01-preview.
+     *
+     * @param apiVersion the apiVersion value.
+     * @return the service client itself
+     */
+    public ProvisioningServiceClientImpl withApiVersion(String apiVersion) {
+        this.apiVersion = apiVersion;
+        return this;
+    }
+
+    /**
+     * Initializes an instance of ProvisioningServiceClient client.
+     */
+    public ProvisioningServiceClientImpl() {
+        this("https://your-dps.azure-devices-provisioning.net");
```

Comment:
I'm a little confused... Who would use this constructor?

Reply:
It's auto-generated from the swagger input file

Reply:
Agree with tim here. This looks odd to stay as a part of the code. Is there any way we can tweak this to not auto generate?

Reply:
It's fetched from here: https://github.com/Azure/azure-iot-sdk-java/blob/dpsServiceSwagger/provisioning/provisioning-service-client/service.json#L8.
We'd have to include this in the swagger discussion to check if it can be removed

Reply:
Is anything needed here in the first place. It looks to me like it should be a comment and left empty especially if we are using it to auto generate code.

Reply:
Yes, that's true; it's possible they updated it from localhost to your-dps.azure-devices-provisioning.net to give better context; we should suggest moving it to a comment instead